pkg/operator/controllers/checker/ingressreplica_test.go

```diff
@@ -0,0 +1,127 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+)
+
+func TestIngressReplicaChecker(t *testing.T) {
+	ctx := context.Background()
+
+	var defaultReplica int32 = 2
+	var zeroReplica int32 = 0
+
+	for _, tt := range []struct {
+		name              string
+		aroCluster        *arov1alpha1.Cluster
+		ingressController *operatorv1.IngressController
+		expectedState     operatorv1.OperatorCondition
+		expectedReplica   int32
+	}{
+		{
+			name: "ingress controller has default replicas",
+			aroCluster: &arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+			},
+			ingressController: &operatorv1.IngressController{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default",
+					Namespace: "openshift-ingress-operator",
+				},
+				Spec: operatorv1.IngressControllerSpec{
+					Replicas: &defaultReplica,
+				},
+			},
+			expectedState: operatorv1.OperatorCondition{
+				Type:    arov1alpha1.DefaultIngressReplica,
+				Status:  operatorv1.ConditionTrue,
+				Message: "Default replicas in place",
+				Reason:  "CheckDone",
+			},
+			expectedReplica: 2,
+		},
+		{
+			name: "ingress controller has zero replicas",
+			aroCluster: &arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+			},
+			ingressController: &operatorv1.IngressController{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default",
+					Namespace: "openshift-ingress-operator",
+				},
+				Spec: operatorv1.IngressControllerSpec{
+					Replicas: &zeroReplica,
+				},
+			},
+			expectedState: operatorv1.OperatorCondition{
+				Type:    arov1alpha1.DefaultIngressReplica,
+				Status:  operatorv1.ConditionTrue,
+				Message: "Rescale succeeded",
+				Reason:  "CheckDone",
+			},
+			expectedReplica: 1,
+		},
+	} {
+		arocli := arofake.NewSimpleClientset()
+		operatorcli := operatorfake.NewSimpleClientset()
+
+		if tt.aroCluster != nil {
+			arocli = arofake.NewSimpleClientset(tt.aroCluster)
+		}
+
+		if tt.ingressController != nil {
+			operatorcli = operatorfake.NewSimpleClientset(tt.ingressController)
+		}
+
+		replicaChecker := NewIngressReplicaChecker(arocli, operatorcli, "")
+
+		t.Run(tt.name, func(t *testing.T) {
+			err := replicaChecker.Check(ctx)
+			if err != nil {
+				t.Error(err)
+			}
+
+			ingress, err := operatorcli.OperatorV1().IngressControllers("openshift-ingress-operator").Get(ctx, "default", metav1.GetOptions{})
+			if err != nil {
+				t.Error(err)
+			}
+
+			if *ingress.Spec.Replicas != tt.expectedReplica {
+				t.Errorf("wrong replica count, want: %d, got: %d", tt.expectedReplica, *ingress.Spec.Replicas)
+			}
+
+			cluster, err := arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+			if err != nil {
+				t.Error(err)
+			}
+
```

Comment:
Add debug info for the condition states on all of these:
```suggestion
receivedCondition := cluster.Status.Conditions[0]
if receivedCondition.Message != tt.expectedState.Message {
    t.Errorf("expected message `%v` but got `%v`", receivedCondition.Message, tt.expectedState.Message)
}
```